File: app/src/main/java/com/frama/miserend/hu/database/local/entities/RecentSearch.java
package com.frama.miserend.hu.database.local.entities;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import java.util.Calendar;

/**
 * Created by maczak on 2018. 02. 16..
 */

@Entity(tableName = "recent_searches")
public class RecentSearch {

    @PrimaryKey
    @ColumnInfo(name = "timestamp")
    private long timestamp;

    @ColumnInfo(name = "searchterm")
    private String searchTerm;

    public RecentSearch(String searchTerm) {
        this.searchTerm = searchTerm;
        this.timestamp = Calendar.getInstance().getTimeInMillis();
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }

    public String getSearchTerm() {
        return searchTerm;
    }

    public void setSearchTerm(String searchTerm) {
        this.searchTerm = searchTerm;
    }
}

File: app/src/main/java/com/frama/miserend/hu/database/local/entities/RecentSearch.kt
package com.frama.miserend.hu.database.local.entities

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import java.util.Calendar

/**
 * Created by maczak on 2018. 02. 16..
 */

@Entity(tableName = "recent_searches")
data class RecentSearch(
        @ColumnInfo(name = "searchterm") var searchTerm: String?,
        @PrimaryKey @ColumnInfo(name = "timestamp") var timestamp: Long = Calendar.getInstance().timeInMillis
)
